salt-master/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM centos:7.6.1810
+
+MAINTAINER moonshot-platform <moonshot-platform@scality.com>
+
+# Salt version to use
+ENV SALT_VERSION 2018.3-1
+
+# salt-master console logging log level
+ENV SALT_LOG_LEVEL warning
+
+# Install saltstack repository
+RUN yum install -y https://repo.saltstack.com/yum/redhat/salt-repo-${SALT_VERSION}.el7.noarch.rpm
+
+RUN yum install -y salt-master salt-api
+
+# salt-master, salt-api
+EXPOSE 4505 4506
+
+CMD /usr/bin/salt-master --log-level=${SALT_LOG_LEVEL}
```

Comment:
To be tested, but I fear that it will leave a lot of zombie process if a proper init 1 process is not deployed first.

I saw that a lot on packaging tests.
I used to use supervisord but maybe we want something lighter here.
```
[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)
logfile_maxbytes=50MB       ; (max main logfile bytes b4 rotation;default 50MB)
logfile_backups=10          ; (num of main logfile rotation backups;default 10)
loglevel=info               ; (logging level;default info; others: debug,warn)
pidfile=/var/run/supervisord.pid ; (supervisord pidfile;default supervisord.pid)
nodaemon=true              ; (start in foreground if true;default false)
minfds=1024                 ; (min. avail startup file descriptors;default 1024)
minprocs=200                ; (min. avail process descriptors;default 200)

;nocleanup=true              ; (don't clean up tempfiles at start;default false)
;http_username=user          ; (default is no username (open system))
;http_password=123           ; (default is no password (open system))
;childlogdir=/tmp            ; ('AUTO' child log dir, default $TEMP)
;user=chrism                 ; (default is current user, required if root)
;directory=/tmp              ; (default is not to cd during start)
;environment=KEY=value       ; (key value pairs to add to environment)

[unix_http_server]
file=/var/run/supervisor.sock
username=setup
password=setup

[supervisorctl]
serverurl=unix:///var/run/supervisor.sock ; use a unix:// URL  for a unix socket
;serverurl=http://127.0.0.1:9001 ; use an http:// url to specify an inet socket
username=setup
password=setup
;prompt=mysupervisor         ; cmd line prompt (default "supervisor")

; the below section must remain in the config file for RPC
; (supervisorctl/web interface) to work, additional interfaces may be
; added by defining them in separate rpcinterface: sections
[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[program:salt-master]
command=/usr/bin/salt-master -l debug
```

Reply:
:100: though not entirely sure about the configuration. We *don't* want `supervisord` to restart `salt-master` on exit etc.

Reply:
So, maybe `supervisord` is not the right approach: http://supervisord.org/index.html?highlight=init

Reply:
Done using `tini`: https://github.com/krallin/tini

Reply:
Fine for now. Will need to change over time, because now we have 'some random binary' in the image :)

Reply:
https://github.com/scality/metalk8s/issues/599